srml/contracts/src/rent.rs

```diff
@@ -165,6 +169,9 @@ fn try_evict_or_and_pay_rent<T: Trait>(
 			..contract
 		});
 
+		let imbalance = imbalance.expect("Imbalance has been checked above");
```

Comment:
I am not sure if I am following the proof. Where exactly it's been checked? What prevents it from being `None` here?

Reply:
I should clarify that. `imbalance` is `Some` if `insufficient_rent || pay_rent` is `true`. [Below, on line 142](https://github.com/paritytech/substrate/pull/3857/files#diff-18b5bc312e600d7b4d3c7ea047d935ccL142) `insufficient_rent || !can_withdraw_rent` is checked, and if `true` then it returns, so when `pay_rent` is checked below, `imbalance` has to be `Some`.

Reply:
I've clarified the proof:
```
imbalance is Some if can_withdraw_rent && (insufficient_rent || pay_rent) is true;
if can_withdraw_rent is false, the function has returned;
pay_rent is true;
qed
```

Reply:
nope, nevermind, I was right the first time.